File: runtime-go/internal/tools/think.go
package tools

import "github.com/ThanabordeeN/AgentRD2/runtime-go/internal/domain"

// thinkSchema mirrors “THINK_SCHEMA“.
var thinkSchema = map[string]any{
	"type": "object",
	"properties": map[string]any{
		"style": map[string]any{
			"type":    "string",
			"enum":    []any{"neutral", "think", "ponder", "listen", "alert", "scheme"},
			"default": "think",
		},
		"duration": map[string]any{
			"type": "number", "minimum": 0.2, "maximum": 30.0, "default": 2.5,
		},
	},
	"additionalProperties": false,
}

// think mirrors “runtime/tools/think.py::think“. Both arguments have Python
// signature defaults ("think", 2.5), and the schema advertises the same
// values.
func think(ctx *Context, args map[string]any) domain.ToolResult {
	style, ok := stringArg(args, "style")
	if !ok {
		style = "think"
	}
	duration, ok := numberArg(args, "duration")
	if !ok {
		duration = 2.5
	}
	return ctx.DispatchAction("think", map[string]any{"style": style, "duration": duration})
}

// RegisterThinkTools registers the thinking family (1 tool).
func RegisterThinkTools(registry *Registry) {
	_ = registry.Register(Spec{
		Name: "think",
		Description: "Play a short thinking/listening/waiting gesture while a decision " +
			"or speech is being prepared.",
		Parameters: thinkSchema,
		Category:   "thinking",
		Handler:    think,
	})
}

File: runtime/tools/think.py
"""Internal thinking / waiting gestures."""
from __future__ import annotations

from typing import Any, Dict, Optional

from runtime.schemas import ToolResult
from runtime.tools.registry import ToolContext, ToolRegistry, ToolSpec


THINK_SCHEMA: Dict[str, Any] = {
    "type": "object",
    "properties": {
        "style": {
            "type": "string",
            "enum": ["neutral", "think", "ponder", "listen", "alert", "scheme"],
            "default": "think",
        },
        "duration": {"type": "number", "minimum": 0.2, "maximum": 30.0, "default": 2.5},
    },
    "additionalProperties": False,
}


def think(
    context: ToolContext,
    style: str = "think",
    duration: float = 2.5,
) -> ToolResult:
    """Play a short thinking/waiting gesture.

    The bridge maps ``style`` to a suitable animation or emote.  This is a
    transient action: it is meant to be replaced by the real decision when
    the LLM or TTS finishes.
    """
    return context.dispatch_action("think", {"style": style, "duration": duration})


def register_think_tools(registry: ToolRegistry) -> None:
    registry.register(ToolSpec(
        "think",
        "Play a short thinking/listening/waiting gesture while a decision or speech is being prepared.",
        THINK_SCHEMA,
        think,
        "thinking",
    ))
